app/provisioning_client.go

```diff
@@ -0,0 +1,128 @@
+package app
+
+import (
+	"context"
+
+	pb "go.viam.com/api/provisioning/v1"
+	"go.viam.com/utils/rpc"
+)
+
+// ProvisioningInfo holds provisioning info.
+type ProvisioningInfo struct {
+	FragmentID   string
+	Model        string
+	Manufacturer string
+}
+
+func provisioningInfoFromProto(info *pb.ProvisioningInfo) *ProvisioningInfo {
+	return &ProvisioningInfo{
+		FragmentID:   info.FragmentId,
+		Model:        info.Model,
+		Manufacturer: info.Manufacturer,
+	}
+}
+
+// NetworkInfo holds network information.
+type NetworkInfo struct {
+	Type      string
+	SSID      string
+	Security  string
+	Signal    int32
+	Connected bool
+	LastError string
+}
+
+func networkInfoFromProto(info *pb.NetworkInfo) *NetworkInfo {
+	return &NetworkInfo{
+		Type:      info.Type,
+		SSID:      info.Ssid,
+		Security:  info.Security,
+		Signal:    info.Signal,
+		Connected: info.Connected,
+		LastError: info.LastError,
+	}
+}
+
+// GetSmartMachineStatusResponse contains smart machine status information.
+type GetSmartMachineStatusResponse struct {
+	ProvisioningInfo           *ProvisioningInfo
+	HasSmartMachineCredentials bool
+	IsOnline                   bool
+	LastestConnectionAttempt   *NetworkInfo
+	Errors                     []string
+}
+
+func getSmartMachineStatusResponseFromProto(resp *pb.GetSmartMachineStatusResponse) *GetSmartMachineStatusResponse {
+	return &GetSmartMachineStatusResponse{
+		ProvisioningInfo:           provisioningInfoFromProto(resp.ProvisioningInfo),
+		HasSmartMachineCredentials: resp.HasSmartMachineCredentials,
+		IsOnline:                   resp.IsOnline,
+		LastestConnectionAttempt:   networkInfoFromProto(resp.LatestConnectionAttempt),
+		Errors:                     resp.Errors,
+	}
+}
+
+// CloudConfig is the minimal config to create a /etc/viam.json, containing the smart machine's part ID and secret.
+type CloudConfig struct {
+	ID         string
+	Secret     string
+	AppAddress string
+}
+
+func cloudConfigToProto(config *CloudConfig) *pb.CloudConfig {
+	return &pb.CloudConfig{
+		Id:         config.ID,
+		Secret:     config.Secret,
+		AppAddress: config.AppAddress,
+	}
+}
+
+// ProvisioningClient is a gRPC client for method calls to the Provisioning API.
+type ProvisioningClient struct {
+	client pb.ProvisioningServiceClient
+}
+
+// NewProvisioningClient constructs a new ProvisioningClient using the connection passed in by the ViamClient.
+func NewProvisioningClient(conn rpc.ClientConn) *ProvisioningClient {
```

Comment:
it looks like I missed this with the previous wrappers (sorry!), but I suspect we want the `New{Foo}Client` methods to be private and the `ViamClient` to be the expected means of accessing the various app clients.